src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -344,9 +459,51 @@ export abstract class BaseCameraPointersInput implements ICameraInput<Camera> {
     protected onLostFocus(): void {
     }
 
+    /**
+     * Called on JS contextmenu event.
+     * Override this method to provide functionality.
+     */
+    protected onContextMenu(evt: PointerEvent): void {
+        evt.preventDefault();
+    }
+
+    /**
+     * This is the queue of events that feeds checkInputs().
+     * See the comment in that function for a description.
+     */
+    private _allEvents: _Event[] = [];
+
     private _pointerInput: (p: PointerInfo, s: EventState) => void;
     private _observer: Nullable<Observer<PointerInfo>>;
     private _onLostFocus: Nullable<(e: FocusEvent) => any>;
-    private pointA: Nullable<PointerTouch>;
-    private pointB: Nullable<PointerTouch>;
+    private _pointA: Nullable<PointerTouch>;
+    private _pointB: Nullable<PointerTouch>;
+}
+
+/**
+ * The following types are for various event types in the queue that feeds
+ * BaseCameraPointersInput.checkInputs().
+ */
+type _TouchEvent = [Nullable<PointerTouch>, number, number];
```

Comment:
I am not sure about the structure of those events. Using an array instead of an objects makes it feel very unstructured.
setting this to be a typescript interface will make it much cleaner:

```javascript
export interface ICameraInputTouchEvent {
  pointerTouch?: PointerTouch;
  name1: number;
  name2: number;
}
```

I know this means restructuring the way the events are stored and the way they work, but it can be unreadable for future devs maintaining this.

Same thing goes to the rest of the events defined (notice that you are not actually using them when creating them, only casting them afterwards)

Reply:
Ack. makes sense.
Typescript is new to me so still trying to get a feeling for what patterns work.

And readability trumps everything IMO.